test/extended/openstack/loadbalancers.go

```diff
@@ -0,0 +1,239 @@
+		depName := "udp-lb-default-dep"
+		svcName := "udp-lb-default-svc"
+		testDeployment := e2edeployment.NewDeployment(depName, 2, labels, "udp-test",
+			imageutils.GetE2EImage(imageutils.Agnhost), appsv1.RollingUpdateDeploymentStrategyType)
+		testDeployment.Spec.Template.Spec.SecurityContext = e2epod.GetRestrictedPodSecurityContext()
+		testDeployment.Spec.Template.Spec.Containers[0].SecurityContext = e2epod.GetRestrictedContainerSecurityContext()
+		testDeployment.Spec.Template.Spec.Containers[0].Args = []string{"netexec", "--udp-port=8081"}
+		testDeployment.Spec.Template.Spec.Containers[0].Ports = []v1.ContainerPort{{
+			ContainerPort: 8081,
+			Protocol:      "UDP",
+		}}
+		deployment, err := clientSet.AppsV1().Deployments(oc.Namespace()).Create(context.TODO(),
+			testDeployment, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = e2edeployment.WaitForDeploymentComplete(clientSet, deployment)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		jig := e2eservice.NewTestJig(clientSet, oc.Namespace(), svcName)
+		jig.Labels = labels
+		svc, err := jig.CreateLoadBalancerService(120*time.Second, func(svc *v1.Service) {
+			svc.Spec.Ports = []v1.ServicePort{{Protocol: v1.ProtocolUDP, Port: 8082, TargetPort: intstr.FromInt(8081)}}
+			svc.Spec.Selector = labels
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("Checks from openshift perspective")
+		o.Expect(svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]).ShouldNot(o.BeEmpty(),
+			"load-balancer-id annotation missing")
+		loadBalancerId := svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]
+		o.Expect(svc.Status.LoadBalancer.Ingress[0].IP).ShouldNot(o.BeEmpty(), "FIP missing on svc Status")
+		o.Expect(svc.Spec.ExternalTrafficPolicy).Should(o.Equal(v1.ServiceExternalTrafficPolicyTypeCluster),
+			"Unexpected ExternalTrafficPolicy on svc specs")
+
+		g.By("Checks from openstack perspective")
+		lb, err := octavialoadbalancers.Get(loadBalancerClient, loadBalancerId).Extract()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(lb.Provider).Should(o.Equal("amphora"), "Unexpected provider in the Openstack LoadBalancer")
+
+		svcIp := svc.Status.LoadBalancer.Ingress[0].IP
+		svcPort := fmt.Sprint(svc.Spec.Ports[0].Port)
+		fip, err := getFipbyFixedIP(networkClient, lb.VipAddress)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(fip.FloatingIP).Should(o.Equal(svcIp), "Unexpected floatingIp in the Openstack LoadBalancer")
+		o.Expect(lb.Pools).Should(o.HaveLen(1), "Unexpected number of pools on Openstack LoadBalancer %q", lb.Name)
+
+		pool, err := pools.Get(loadBalancerClient, lb.Pools[0].ID).Extract()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(pool.Protocol).Should(o.Equal("UDP"), "Unexpected protocol on Openstack LoadBalancer Pool: %q", pool.Name)
+		o.Expect(pool.ProvisioningStatus).Should(o.Equal("ACTIVE"), "Unexpected Provisioning Status on Openstack Pool: %q", pool.Name)
```

Comment:
This check might be problematic, what if a new node just got created and added as LB member, so that pool is in PENDING_UPDATE? This is perfectly valid behavior, yet would make test fails. I'd rather skip this one.

Reply:
Removed, thanks!